runtime/memory.c

```diff
@@ -262,6 +262,8 @@ char *caml_alloc_for_heap (asize_t request)
     mem = (char *) block + sizeof (heap_chunk_head);
     Chunk_size (mem) = size - sizeof (heap_chunk_head);
     Chunk_block (mem) = block;
+    Chunk_redarken_start(mem) = (value*)(mem + size);
+    Chunk_redarken_end(mem) = 0;
```

Comment:
There is a strange asymmetry in this initialization.  I would have expected
```
Chunk_redarken_end(mem) = mem;
```
if you want both start and end pointers to remain within the malloc-ed block, which is good to ensure well-definedness of comparisons later.  If you know that pointers are compared as unsigned integers, you could do
```
    Chunk_redarken_start(mem) = (value*)(-1);
    Chunk_redarken_end(mem) = 0;
```

Reply:
Yes, this is a much clearer approach. I have changed this in the most recent commit.

Reply:
I have second thoughts about the code I suggested :-(  If pointers are wider than type `int`, it's not specified whether the value `-1` (of type `int`) will be zero- or sign-extended to a bit pattern as wide as a pointer, as @dra27 pointed out recently.  In the former case (zero-extension) we don't get the "all ones" bit pattern we wanted.  It would be safer to write
```
Chunk_redarken_start(mem) = (value*) UINTPTR_MAX;
Chunk_redarken_end(mem) = (value *) 0;
```
This said, we're still assuming that pointers are compared as unsigned integers, which is a popular choice among compiler writers but not a guarantee.  And the comparison will be undefined behavior (strictly speaking).  So in the end I would prefer the first suggestion:
```
Chunk_redarken_start(mem) = (value*)(mem + actual_size);
Chunk_redarken_end(mem) = (value *) mem;
```

Reply:
This should be fixed in the latest commit.